subprojects/kotlin-dsl/kotlin-dsl.gradle.kts

```diff
@@ -143,3 +144,39 @@ tasks {
         from(writeVersionsManifest)
     }
 }
+
+
+// -- Embedded Kotlin dependencies -------------------------------------
+
+val embeddedKotlinBaseDependencies by configurations.creating
+
+dependencies {
+    embeddedKotlinBaseDependencies(futureKotlin("stdlib-jdk8"))
+    embeddedKotlinBaseDependencies(futureKotlin("reflect"))
+}
+
+val writeEmbeddedKotlinDependencies by tasks.registering {
+    val outputFile = layout.buildDirectory.file("embeddedKotlinDependencies/gradle-kotlin-dsl-embedded-kotlin.properties")
+    inputs.files(embeddedKotlinBaseDependencies)
+    outputs.file(outputFile)
+    doLast {
+
+        val modules = embeddedKotlinBaseDependencies.incoming.resolutionResult.allComponents
+            .asSequence()
+            .mapNotNull { it.moduleVersion }
+            .filter { it.name !in setOf(project.name, "distributionsDependencies", "kotlinCompilerEmbeddable") }
```

Comment:
❌ Hoist `setOf(...)` expression to make chain a little cheaper

Reply:
Done